cmd/secret-manager/util.py

```
@@ -0,0 +1,370 @@
+    try:
+        # Resolve the path and check if it's readable
+        resolved_path = os.path.realpath(from_file)
+        if not os.path.isfile(resolved_path):
+            raise click.UsageError(
+                f"File '{from_file}' does not exist or is not a regular file."
+            )
+
+        with open(resolved_path, "rb") as f:
+            return f.read()
+    except (OSError, IOError) as e:
+        raise click.UsageError(f"Failed to read file '{from_file}': {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)